qiskit/utils/deprecation.py

```diff
@@ -89,27 +102,85 @@ def wrapper(*args, **kwargs):
 def _rename_kwargs(
     func_name: str,
     kwargs: Dict[str, Any],
-    kwarg_map: Dict[str, str],
+    old_kwarg_to_msg: Dict[str, str],
+    kwarg_map: Dict[str, Optional[str]],
     category: Type[Warning] = DeprecationWarning,
 ) -> None:
     for old_arg, new_arg in kwarg_map.items():
-        if old_arg in kwargs:
-            if new_arg in kwargs:
-                raise TypeError(f"{func_name} received both {new_arg} and {old_arg} (deprecated).")
-
-            if new_arg is None:
-                warnings.warn(
-                    f"{func_name} keyword argument {old_arg} is deprecated and "
-                    "will in future be removed.",
-                    category=category,
-                    stacklevel=3,
-                )
-            else:
-                warnings.warn(
-                    f"{func_name} keyword argument {old_arg} is deprecated and "
-                    f"replaced with {new_arg}.",
-                    category=category,
-                    stacklevel=3,
-                )
-
-                kwargs[new_arg] = kwargs.pop(old_arg)
+        if old_arg not in kwargs:
+            continue
+        if new_arg in kwargs:
+            raise TypeError(f"{func_name} received both {new_arg} and {old_arg} (deprecated).")
+        warnings.warn(old_kwarg_to_msg[old_arg], category=category, stacklevel=3)
+        if new_arg is not None:
+            kwargs[new_arg] = kwargs.pop(old_arg)
+
+
+def _add_deprecation_to_docstring(
+    func: Callable, msg: str, *, since: Optional[str], pending: bool
+) -> None:
+    if since is None:
+        version_str = "unknown"
+    else:
+        version_str = f"{since}_pending" if pending else since
+
+    # We insert deprecations in-between the description and the meta sections. The below is from
+    # https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html#docstring-sections.
+    # We only include what is used by Qiskit projects for functions/methods to avoid wasting
+    # computation.
```

Comment:
I should probably just add them all? Lookup in a set is very fast, and not impacted much by a few extra elements.

Reply:
This impacts imports of Terra too. But it should be pretty fast because the algorithm is efficient

Reply:
We should add them all - I can see at the very least that "See also" isn't in this list, and I know that's used somewhere, but mostly we shouldn't need to update this function based on text changes we'd reasonably expect to work.  If you're concerned about runtime performance, we could reduce the set-construction cost (which will be more expensive than the lookups) by making it a private module-level constant instead.